### PR TITLE
[AV2-263] Feature/instant login

### PR DESCRIPTION
**Pull request by trtshen**

### CODE REVIEW

**Comment by trtshen** [comment]
> image
> It's better to put the background image above the text & input & button. Now it looks a bit weird.

@shawnm0705 isn't it the background image is above them all now?

File: src/app/go-mobile/go-mobile.component.html [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+                    <ion-select placeholder="Select Country" name="countryModel" [(ngModel)]="countryModel" (ionChange)="updateCountry()">
+                        <ion-select-option *ngFor="let countryCode of countryCodes" [value]="countryCode.code">{{countryCode.name}}</ion-select-option>
+                    </ion-select>
+                </ion-col>
+            </ion-item>
+        </div>
+        <div text-center>
+            <ion-button expand="block" type="submit" [disabled]="invalidNumber || loading">Send Instant-Login via SMS  <ion-spinner *ngIf="loading"></ion-spinner></ion-button>
+
+            <a href="/app/home">not right now...</a>
```
[the remark this answers is not in the document]
Comment (trtshen) [reply src/app/go-mobile/go-mobile.component.html]: im ok with any one, i'll change it to `/switch`

File: src/app/go-mobile/go-mobile.component.ts [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+import { environment } from '../../environments/environment.prod';
+
+@Component({
+  selector: 'app-switcher',
+  templateUrl: './go-mobile.component.html',
+  styleUrls: ['./go-mobile.component.scss']
+})
+export class GoMobileComponent implements OnInit {
+  loading = false;
+  saved = false;
```
[the remark this answers is not in the document]
Comment (trtshen) [reply src/app/go-mobile/go-mobile.component.ts]: the differences between them are `two-way binding` and `submission status`

**Comment by shawnm0705** [comment]
It is currently overing them. See the purple circle at bottom

**Comment by shawnm0705** [comment]
> > image
> > It's better to put the background image above the text & input & button. Now it looks a bit weird.
> 
> @shawnm0705 isn't it the background image is above them all now?

It is currently overing them. See the purple circle at bottom

File: src/app/go-mobile/go-mobile.component.ts [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+    private notification: NotificationService,
+    public storage: BrowserStorageService,
+    private contact: ContactNumberFormat,
+  ) {
+    this.countryCodes = contact.countryCodes;
+    this.formatMasks = contact.masks;
+  }
+
+  ngOnInit() {
+    this.profile.email = this.storage.getUser().email;
```
[the remark this answers is not in the document]
Comment (trtshen) [reply src/app/go-mobile/go-mobile.component.ts]: actually the api works even without the email, i'll remove it instead

File: src/app/auth/auth.service.ts [hunk 4]
```diff
@@ -1,8 +1,8 @@
 import { Injectable } from '@angular/core';
 import { RequestService, QueryEncoder } from '@shared/request/request.service';
 import { HttpParams } from '@angular/common/http';
-import { map } from 'rxjs/operators';
-import { Observable, of } from 'rxjs';
+import { map, flatMap } from 'rxjs/operators';
+import { Observable, of, concat } from 'rxjs';
```
Comment (shawnm0705) [inline src/app/auth/auth.service.ts]: `flatMap` and `concat` are not being used

File: src/app/go-mobile/go-mobile.component.ts [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+  countryCodes: any[];
+
+  constructor(
+    private goMobileService: GoMobileService,
+    private utils: UtilsService,
+    private notification: NotificationService,
+    public storage: BrowserStorageService,
+    private contact: ContactNumberFormat,
+  ) {
+    this.countryCodes = contact.countryCodes;
```
Comment (shawnm0705) [inline src/app/go-mobile/go-mobile.component.ts]: Why this comes back again?

File: src/app/switcher/switcher.service.ts [hunk 6]
```diff
@@ -59,7 +60,8 @@ export class SwitcherService {
     private request: RequestService,
     private utils: UtilsService,
     private storage: BrowserStorageService,
-    private sharedService: SharedService
+    private sharedService: SharedService,
+    private authService: AuthService,
```
Comment (shawnm0705) [inline src/app/switcher/switcher.service.ts]: This is not used

File: src/app/go-mobile/go-mobile.component.ts (hunk 5 above)
Comment (trtshen) [reply src/app/go-mobile/go-mobile.component.ts]: @shawnm0705 needs this bro
https://github.com/intersective/practera-app-v2/pull/318/files/70e284c09c7fe9dc47841273fd0fb89b7b5c79b7#diff-b900146e0fbc39787e4b5bd6bd80d534R40

Comment (shawnm0705) [reply src/app/go-mobile/go-mobile.component.ts]: @trtshen We can use `contact.countryCodes` there as well

File: src/app/go-mobile/go-mobile.component.ts [hunk 7]
```diff
@@ -0,0 +1,137 @@
+import { Component, OnInit } from '@angular/core';
+import { ModalController } from '@ionic/angular';
+import { GoMobileService } from './go-mobile.service';
+import { FormGroup, FormControl, Validators } from '@angular/forms';
+import { UtilsService, ContactNumberFormat } from '@services/utils.service';
+import { NotificationService } from '@shared/notification/notification.service';
+import { BrowserStorageService } from '@services/storage.service';
+import { environment } from '../../environments/environment.prod';
```
[the remark this answers is not in the document]
Comment (trtshen) [reply src/app/go-mobile/go-mobile.component.ts]: I'll change to default `environment.ts`
cc: @jazzmind

**Pull request merged**